pybind11_stubgen/__init__.py

```diff
@@ -221,6 +227,26 @@ def remove_signatures(docstring):  # type: (str) ->str
 
         return "\n".join(filter(lambda line: not re.match(signature_regex, line), lines))
 
+    @staticmethod
+    def remove_repeated_empty_lines(docstring):
+        # Reduce the double empty line to a single empty line
+        return re.sub(r"\n\s*\n", "\n\n", docstring)
+
+    @staticmethod
+    def sanitize_docstring(docstring):  # type: (str) ->str
+        docstring = StubsGenerator.remove_signatures(docstring)
+        docstring = StubsGenerator.remove_repeated_empty_lines(docstring)
```

Comment:
I doubt about removing repeated empty lines. The format of docstring is up to library author, probably we should limit our-self to trimming outer empty lines only, see https://www.python.org/dev/peps/pep-0257/#id18 

The docstring indentation can be managed in another PR

Reply:
True, we shouldn't force our rules. The reason why I implemented it is by removing the signatures, we create these double empty lines.

We should actually fix that problem. So that we remove the signature line, including the first or previous empty line.

PS, I experienced two problems with the signature removing.
1. It can't handle any brackets in the signature, this could possibly happen in the default value representation.
2. It can't handle new lines in the signature. This is caused by splitting on "\n". Signatures are always separated by an empty line, the one we need to remove, so splitting on "\n\n" would maybe fix both problems in once.

Reply:
Splitting on "\n\n" (and again joining on "\n\n"), does fix the problem of creating double empty lines, when remove signatures. Therefore there is no need to remove the double lines anymore, which would also change the original spacing in the remainder of the docstring.